common/src/main/kotlin/org/neo4j/connectors/kafka/data/PropertyType.kt

```diff
@@ -0,0 +1,366 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.neo4j.connectors.kafka.data
+
+import java.nio.ByteBuffer
+import java.time.LocalDate
+import java.time.LocalDateTime
+import java.time.LocalTime
+import java.time.OffsetDateTime
+import java.time.OffsetTime
+import java.time.ZoneOffset
+import java.time.ZonedDateTime
+import java.time.format.DateTimeFormatter
+import java.time.temporal.TemporalQueries
+import kotlin.reflect.KClass
+import org.apache.kafka.connect.data.Schema
+import org.apache.kafka.connect.data.SchemaBuilder
+import org.apache.kafka.connect.data.Struct
+import org.neo4j.driver.Values
+import org.neo4j.driver.types.IsoDuration
+import org.neo4j.driver.types.Point
+
+@Suppress("UNCHECKED_CAST")
+object PropertyType {
+  internal const val MONTHS = "months"
+  internal const val DAYS = "days"
+  internal const val SECONDS = "seconds"
+  internal const val NANOS = "nanoseconds"
+  internal const val SR_ID = "srid"
+  internal const val X = "x"
+  internal const val Y = "y"
+  internal const val Z = "z"
+  internal const val DIMENSION = "dimension"
+  internal const val TWO_D: Byte = 2
+  internal const val THREE_D: Byte = 3
+
+  internal const val BOOLEAN = "B"
+  internal const val BOOLEAN_LIST = "LB"
+  internal const val LONG = "I64"
+  internal const val LONG_LIST = "LI64"
+  internal const val FLOAT = "F64"
+  internal const val FLOAT_LIST = "LF64"
+  internal const val STRING = "S"
+  internal const val STRING_LIST = "LS"
+  internal const val BYTES = "BA"
+  internal const val LOCAL_DATE = "TLD"
+  internal const val LOCAL_DATE_LIST = "LTLD"
+  internal const val LOCAL_DATE_TIME = "TLDT"
+  internal const val LOCAL_DATE_TIME_LIST = "LTLDT"
+  internal const val LOCAL_TIME = "TLT"
+  internal const val LOCAL_TIME_LIST = "LTLT"
+  internal const val ZONED_DATE_TIME = "TZDT"
+  internal const val ZONED_DATE_TIME_LIST = "LZDT"
+  internal const val OFFSET_TIME = "TOT"
+  internal const val OFFSET_TIME_LIST = "LTOT"
+  internal const val DURATION = "TD"
+  internal const val DURATION_LIST = "LTD"
+  internal const val POINT = "SP"
+  internal const val POINT_LIST = "LSP"
+
+  internal val durationSchema: Schema =
+      SchemaBuilder(Schema.Type.STRUCT)
+          .field(MONTHS, Schema.INT64_SCHEMA)
+          .field(DAYS, Schema.INT64_SCHEMA)
+          .field(SECONDS, Schema.INT64_SCHEMA)
+          .field(NANOS, Schema.INT32_SCHEMA)
+          .optional()
+          .build()
+
+  internal val pointSchema: Schema =
+      SchemaBuilder(Schema.Type.STRUCT)
+          .field(DIMENSION, Schema.INT8_SCHEMA)
+          .field(SR_ID, Schema.INT32_SCHEMA)
+          .field(X, Schema.FLOAT64_SCHEMA)
+          .field(Y, Schema.FLOAT64_SCHEMA)
+          .field(Z, Schema.OPTIONAL_FLOAT64_SCHEMA)
+          .optional()
+          .build()
+
+  val schema: Schema =
+      SchemaBuilder.struct()
+          .namespaced("Neo4jPropertyType")
```

Comment:
do we need a version field in there in case some new types land in newer Neo4j releases?

Reply:
Adding new fields are allowed as far as we add them as `nullable` so I don't think we need an explicit version field here.